src/Analyzer/Passes/AutoFinalOnQueryPass.cpp

```diff
@@ -42,7 +42,7 @@ class AutoFinalOnQueryPassVisitor : public InDepthQueryTreeVisitorWithContext<Au
             return;
 
         const auto & storage = table_node ? table_node->getStorage() : table_function_node->getStorage();
-        bool is_final_supported = storage && storage->supportsFinal();
+        bool is_final_supported = storage && !storage->isRemote() && storage->supportsFinal();
```

Comment:
Just curious: how did it work in old infrastructure?

Reply:
Same. I have copied from this
https://github.com/ClickHouse/ClickHouse/blob/51afec49107864e97eb36f9e5760efd1e11bfea8/src/Interpreters/InterpreterSelectQuery.cpp#L3268